Fix PostAttributeChangeCallback value ctypes typing

sasdf opened the pull request:
This PR fixes the incorrect PostAttributeChangeCallback's value argument ctypes typing, which is documented in the TODO.
>   TODO: This should be a pointer to uint8_t, but ctypes does not provide
>         such a type. The best approximation is c_char_p, however, this
>         requires the caller to pass NULL-terminate C-string which might
>         not be the case here.

The callback requires the caller to pass a NULL-terminated C-string, but it’s actually a length-buffer pair. The python lighting example receives values with incorrect length due to this issue.

With this PR, the `PostAttributeChangeCallback` decorator will wrap the callback function, and converts the length-buffer pair to a proper python bytes value.

To keep it backward-compatible, the length argument will still be passed to the callback, although it becomes redundant.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=36928) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=36928) before we can accept your contribution.<br/><hr/>**sasdf** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/project-chip/connectedhomeip?pullRequest=36928) it.</sub>

sasdf committed:
Fix PostAttributeChangeCallback value typing

As documented in the TODO, the callback requires the caller to pass a
NULL-terminated C-string, but it’s actually a length-buffer pair.
The python lighting example receives values with incorrect length due to this
issue.

This PR fixes the FFI interface by wrapping the callback function, and converts
the length-buffer pair to a proper python bytes value.
